piptools/scripts/compile.py

```diff
@@ -328,13 +363,21 @@ def cli(
     emit_index_url: bool,
     emit_options: bool,
     unsafe_package: tuple[str, ...],
+    build_distributions: tuple[str, ...],
+    all_build_distributions: bool,
+    only_build_distributions: bool,
 ) -> None:
     """
     Compiles requirements.txt from requirements.in, pyproject.toml, setup.cfg,
     or setup.py specs.
     """
     log.verbosity = verbose - quiet
 
+    if only_build_distributions and any([extras, all_extras]):
```

Comment:
`any()` is not really needed here, it only complicates things
```suggestion
    if only_build_distributions and (extras or all_extras):
```

Reply:
`any()` is not really needed here, it only complicates things
```suggestion
    any_extras_requested = extras or all_extras
    if only_build_distributions and any_extras_requested:
```

Reply:
Good point.

Reply:
Looks like this has been rewritten. Marking the conversation as resolved.